src/packet.c

```diff
@@ -1,3 +1,209 @@
+        swap_16bit(&reply->num_subs);
+      } break;
+      case BCMP_NEIGHBOR_PROTO_REQUEST: {
+        BCMPNeighborProtoRequest *request = (BCMPNeighborProtoRequest *)buf;
+        swap_64bit(&request->target_node_id);
+        swap_16bit(&request->option_count);
+      } break;
+      case BCMP_NEIGHBOR_PROTO_REPLY: {
+        BCMPNeighborProtoReply *request = (BCMPNeighborProtoReply *)buf;
+        swap_64bit(&request->node_id);
+        swap_16bit(&request->option_count);
+      } break;
+      case BCMP_SYSTEM_TIME_REQUEST: {
+        BCMPSystemTimeRequest *request = (BCMPSystemTimeRequest *)buf;
+        swap_64bit(&request->header.target_node_id);
+        swap_64bit(&request->header.source_node_id);
+      } break;
+      case BCMP_SYSTEM_TIME_RESPONSE: {
+        BCMPSystemTimeResponse *response = (BCMPSystemTimeResponse *)buf;
+        swap_64bit(&response->header.target_node_id);
+        swap_64bit(&response->header.source_node_id);
+        swap_64bit(&response->utc_time_us);
+      } break;
+      case BCMP_SYSTEM_TIME_SET: {
+        BCMPSystemTimeSet *set = (BCMPSystemTimeSet *)buf;
+        swap_64bit(&set->header.target_node_id);
+        swap_64bit(&set->header.source_node_id);
+        swap_64bit(&set->utc_time_us);
+      } break;
+      case BCMP_NET_STAT_REQUEST: {
+        BCMPNetstatRequest *request = (BCMPNetstatRequest *)buf;
+        swap_64bit(&request->target_node_id);
+      } break;
+      case BCMP_NET_STAT_REPLY: {
+        BCMPNetstatReply *reply = (BCMPNetstatReply *)buf;
+        swap_64bit(&reply->node_id);
+      } break;
+      case BCMP_POWER_STAT_REQUEST: {
+        BCMPPowerStateRequest *request = (BCMPPowerStateRequest *)buf;
+        swap_64bit(&request->target_node_id);
+      } break;
+      case BCMP_POWER_STAT_REPLY: {
+        BCMPPowerStateReply *reply = (BCMPPowerStateReply *)buf;
+        swap_64bit(&reply->node_id);
+      } break;
+      case BCMP_REBOOT_REQUEST: {
+        BCMPRebootRequest *request = (BCMPRebootRequest *)buf;
+        swap_64bit(&request->target_node_id);
+      } break;
+      case BCMP_REBOOT_REPLY: {
+        BCMPRebootReply *reply = (BCMPRebootReply *)buf;
+        swap_64bit(&reply->node_id);
+      } break;
+      case BCMP_NET_ASSERT_QUIET: {
+        BCMPNetAssertQuiet *request = (BCMPNetAssertQuiet *)buf;
+      } break;
+      case BCMP_CONFIG_GET:
+      case BCMP_CONFIG_VALUE:
+      case BCMP_CONFIG_SET:
+      case BCMP_CONFIG_COMMIT:
+      case BCMP_CONFIG_STATUS_REQUEST:
+      case BCMP_CONFIG_STATUS_RESPONSE:
+      case BCMP_CONFIG_DELETE_REQUEST:
+      case BCMP_CONFIG_DELETE_RESPONSE:
+      case BCMP_DFU_START:
+      case BCMP_DFU_PAYLOAD_REQ:
+      case BCMP_DFU_PAYLOAD:
+      case BCMP_DFU_END:
+      case BCMP_DFU_ACK:
+      case BCMP_DFU_ABORT:
+      case BCMP_DFU_HEARTBEAT:
+      case BCMP_DFU_REBOOT_REQ:
+      case BCMP_DFU_REBOOT:
+      case BCMP_DFU_BOOT_COMPLETE:
+      case BCMP_HEADER:
+        break;
+      }
+    }
```

Comment:
The added switch case is located here.
The only pain I see here is unit test coverage. But I think it will be important to test all message structures anyway in the unit test as that can be an easy way to validate serialization before throwing onto an actual device.

Reply:
Please extract the "if big endian then swap fields" into its own function so that the intent of the serialize function is visually clear. Right now it looks like the point of serialize is to swap endianness, when in reality that will be a super rare code path.

Reply:
Gotcha!